detoks/src/main/java/nl/tudelft/trustchain/detoks/TorrentManager.kt

```diff
@@ -325,7 +325,7 @@ class TorrentManager private constructor (
         }
 
         // Preserve cached if again in cache
-        val cacheEnd = currentIndex + cachingAmount
+        val cacheEnd = Math.min(torrentFiles.size - 1, currentIndex + cachingAmount)
```

Comment:
Instead of this change, for getting items can use `.gett(i)` instead of `[i]` for wrap around. I think there is not a function for setting with wrap around, but `gett(i)` just uses `index.mod(size)`, so then you can do `set(element, i.mod(size))`